### PR TITLE
Add different behaviors to single tap

### PR DESCRIPTION
**Pull request by hyouuu**
Currently when image is zoomed, a single tap has no response, whereas some design may want it to act as zooming back or dismiss (our app actually needs to dismiss on tap), so added the options

### CODE REVIEW

**Comment by hyouuu** [comment]
Ping on this - your thoughts?

**Comment by hyouuu** [comment]
Wow flaky SchnaubBot - it's already weak...

**Comment by JanGorman** [comment]
Yeah no idea what's up with that warning. Swiftlint doesn't report that if run in Xcode. Anyway, sorry for the late reply. I just have some suggestions on naming, otherwise I think this is fine to add

File: Agrume/AgrumeCell.swift [hunk 1]
```diff
@@ -205,8 +205,21 @@ extension AgrumeCell: UIGestureRecognizerDelegate {
   }
 
   @objc
-  private func singleTap() {
-    dismiss()
+  private func singleTap(_ gesture: UITapGestureRecognizer) {
```
Comment (JanGorman) [inline Agrume/AgrumeCell.swift]: No need to pass in the `gesture` parameter, just `singleTap()`

File: Agrume/Agrume.swift [hunk 2]
```diff
@@ -30,6 +30,13 @@ public final class Agrume: UIViewController {
   /// Hide status bar when presenting. Defaults to `false`
   public var hideStatusBar = false
 
+  public enum TapBehavior {
+    case dismissOnlyIfUnzoomed
+    case dismissAlways
+    case unzoomIfZoomed
```
Comment (JanGorman) [inline Agrume/Agrume.swift]: `zoomOut`

File: Agrume/Agrume.swift [hunk 3]
```diff
@@ -30,6 +30,13 @@ public final class Agrume: UIViewController {
   /// Hide status bar when presenting. Defaults to `false`
   public var hideStatusBar = false
 
+  public enum TapBehavior {
+    case dismissOnlyIfUnzoomed
```
Comment (JanGorman) [inline Agrume/Agrume.swift]: `dismissIfZoomedOut`

**Comment by hyouuu** [comment]
Good point - updated!

**Pull request merged**